xbmc/music/MusicDatabase.cpp

```diff
@@ -124,6 +124,7 @@ void CMusicDatabase::CreateTables()
   m_pDS->exec("CREATE TABLE artist ( idArtist integer primary key, "
               " strArtist varchar(256), strMusicBrainzArtistID text, "
               " strSortName text, "
+              " strType text, strGender text, strDisambiguation text, "
```

Comment:
Shouldn't these be added at the very end of the list of columns? Because otherwise you will end up with a different column layout for newly created tables vs. updated tables which will screw up positional indexing on SELECT queries.

Reply:
There is no positinal indexing or access used on the music db because of such a vulnerability. Changing column layout is not an issue for the db in the way it is fro JSON itself

Reply:
Personally I'd prefer if the database layout looks the same for a newly created database and an updated database even if there is no positional indexing right now.

Reply:
That has not happened before so already have differences. There are potentially more changes to come, so I could do an migration step that makes column order same for all.

Reply:
Right I haven't done much in the music database. Feel free to do whatever you prefer.